tests/trainers/java/threads_concurrency_and_sleep.java

```diff
@@ -0,0 +1,51 @@
+/**
+Test to check the Java Threading module.
+It runs two concurrent tests and wait for their ending.
+Also includes a pasive wait (sleep) in one of the thread.
+*/
+public class Test {
+    public static void main(String[] args) {
+        BasicThread t1 = new BasicThread("1");
+        t1.start();
+        BasicThread2 t2 = new BasicThread2("2");
+        t2.start();
+
+        try {
+          t1.join();
+          t2.join();
+        } catch ( Exception e) {
```

Comment:
Catching `Exception` will always work but is disencouraged. One should only catch RuntimeExceptions or more specifically, the actual RuntimeException that are occurring 

```suggestion
        } catch ( InterruptedException e) {
```

I don't think this is really important to do, but the change is also very cheap and should not hurt IMHO :slightly_smiling_face:

Reply:
done